### PR TITLE
@starsirius: initial exploration with integration testing for gris client apps

### PR DESCRIPTION
**Pull request by dylanfareed**
Just a very basic first pass at an integration test that generates a new Gris app and returns what we are expecting. Am curious to see whether this works on Semaphore.

### CODE REVIEW

**Comment by dylanfareed** [comment]
If this feels right and pans out, I'll more properly reorganize. Thanks!

**Comment by starsirius** [comment]
This looks good to me, and Semaphore seems to be happy as well! :+1: So glad to see Sigma's spirit coming back. :P

File: spec/integration/token_authentication_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  context 'with correct token query params' do
+    it 'returns the root presenter' do
+      get "http://localhost:#{app_port}/?token=replace-me"
+      expect_json_keys(:_links)
+    end
+  end
+
+  context 'with correct token query params' do
+    it 'returns the root presenter' do
+      get "http://localhost:#{app_port}/", 'Http-Authorization' => 'replace-me'
```
Comment (dylanfareed) [inline spec/integration/token_authentication_spec.rb]: Very basic specs proving that a newly generated app will behave as expected with the token authentication helper.

File: spec/support/integration_setup.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        start_app
+      rescue ChildProcess::TimeoutError
+        project.stop
+      end
+    end
+  end
+
+  def start_app
+    @app = ChildProcess.build('bundle', 'exec', 'rackup')
+    @app.environment['PERMITTED_TOKENS'] = 'replace-me'
```
Comment (dylanfareed) [inline spec/support/integration_setup.rb]: I am setting the env variable manually here for the newly running app, but we'll want a more flexible path to booting and customizing these for different spec contexts.

**Comment by dylanfareed** [comment]
Lemme know what you think @starsirius. This feels like a good starting point.

File: spec/integration/token_authentication_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+require 'hyperclient'
+
+describe 'integration' do
+  include_context 'with a generated app'
+
+  let(:client_url) { "http://localhost:#{app_port}" }
+  let(:client) { Hyperclient.new(client_url) }
```
Comment (dylanfareed) [inline spec/integration/token_authentication_spec.rb]: Using hyperclient to test the newly spun up API

**Comment by dylanfareed** [comment]
YOLO. Merging.

**Comment by starsirius** [comment]
:+1: Sorry about the delay. This is a great start!

**Pull request merged**